os/arch/arm/src/common/up_createstack.c

```diff
@@ -243,6 +253,16 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
+#ifdef CONFIG_MPU_STACKGUARD
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+		tcb->stack_guard = (uint32_t)tcb->stack_alloc_ptr + 0x40;
+		tcb->stack_guard = ((uint32_t)tcb->stack_guard & STACK_GUARD_ADDR_MASK);
+		tcb->guard_size = ((uint32_t)tcb->stack_guard - (uint32_t)tcb->stack_alloc_ptr);
+		mpu_configure_app_regs(&tcb->mpu_regs[STACKGUARD_MPU_NUM * REGS_PER_REGION], STACKGUARD_MPU_NUM, tcb->stack_guard, tcb->guard_size, true, false);
```

Comment:
Is this added to set MPU region for stack guard? What's the difference between calling this and calling up_set_mpu_region_configuration?

Reply:
mpu_configure_app_regs will just keep the mpu data structures ready in the tcb structure variable.
up_set_mpu_region_configuration will actually set those values in MPU (It will be done during  context switch, scheduling task)